src/app/process-editor/services/cardview-properties/timer-definition-item/timer-definition-item.component.ts

```diff
@@ -0,0 +1,103 @@
+ /*!
+ * @license
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { CardItemTypeService, CardViewUpdateService, AppConfigService } from '@alfresco/adf-core';
+import { FormBuilder, Validators, FormControl, FormGroup, AbstractControl } from '@angular/forms';
+import { debounceTime } from 'rxjs/operators';
+
+@Component({
+    selector: 'ama-process-timer-definition',
+    templateUrl: './timer-definition-item.component.html',
+    providers: [CardItemTypeService]
+})
+export class CardViewTimerDefinitionItemComponent implements OnInit {
+    @Input() property;
+
+    timers = [];
+    selectedTimer: Bpmn.DiagramElement;
+    defaultTimerDefinition = '';
+    defaultTimerType = '';
+
+    timerDefinitionForm: FormGroup;
+
+    constructor(
+        private cardViewUpdateService: CardViewUpdateService,
+        private appConfigService: AppConfigService,
+        private formBuilder: FormBuilder
+    ) {}
+
+    ngOnInit() {
+        this.setTimerFromXML();
+        this.timers = this.appConfigService.get('process-modeler.timer-types');
+
+        this.buildForm();
+    }
+
+    buildForm() {
+        this.timerDefinitionForm = this.formBuilder.group({
+            timerType: new FormControl(this.defaultTimerType, [Validators.required]),
+            timerDefinition: new FormControl(this.defaultTimerDefinition, [Validators.required])
+        });
+
+        this.timerDefinitionForm.valueChanges
+            .pipe(
+                debounceTime(500)
+            )
+            .subscribe((timerDefinitionValues: any) => {
```

Comment:
An observer subscription without unsubscribe. Please unsubscribe it on ngOnDestroy, otherwise we are leaking memory. Like here:
`/src/app/app.component.ts`

Reply:
You still need to use the `this.onDestroy$` here, because simply calling it inside the onDestroy, won't stop this subscription. Please review the linked file once again.
```
.pipe(takeUntil(this.onDestroy$))
```